hack/shellcheck.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -eux
+
+if [ "$IS_CONTAINER" != "" ]; then
+  TOP_DIR="${1:-.}"
+  find "${TOP_DIR}" \
+    -type f -name '*.sh' -exec shellcheck -s bash {} \+
+else
+  podman run --rm \
+    --env IS_CONTAINER=TRUE \
+    --volume "${PWD}:/workdir:ro,z" \
+    --entrypoint sh \
+    --workdir /workdir \
+    koalaman/shellcheck-alpine:stable \
+    /workdir/hack/shellcheck.sh "${@}"
```

Comment:
Is there a reason to not just always do the container? I don't know how often shellcheck gets updated, but I could see a case where things work locally but but the container has a newer version

Reply:
Yeah, when prow runs this in a CI job, it's going to run this script inside the same container image

Reply:
https://github.com/metal3-io/project-infra/pull/6

Reply:
I meant why default to not using the container? I like the container approach, it means everyone’s always running the same code And there’s no need to have the dependencies pre-installed

Reply:
I think it does default to using the container?  Well, it actually fails if you just run it because `IS_CONTAINER` isn't set, but it will use the container if you run `make lint`

Reply:
nevermind this is broken, will fix ..

Reply:
Oh I see, I misunderstood the meaning of `$IS_CONTAINER`, sorry for the noise

Reply:
no problem - it wasn't actually working right, either.  I think it's better now ...